COLE-readthis.md

```diff
@@ -0,0 +1,38 @@
+* Notes
+** I wasn't sure where to put my comments about steps/tasks/instructions in the code because everything is spread out over different files. If I put notes in one file you might not know whether I went there first or second or whatever. So here it is.
+** After reading the README 3 or 4 times I had no idea what I started to get a little upset because I wasn't sure what half of it meant, so I used the checklist that Ellen Nitchals sent out to guide me. Sorry. I know I'm supposed to follow the README but half the time they just confuse me more than I already was and I just go by the checklists as they make a lot more sense and are much easier to follow.
+** Question: single quotes or double quotes?
+
+* Project Organization
+
+** Step 1: npm setup
+*** I understood this part of the README and it's just like a lot of other projects we've done recently so everything went according to plan except for the last part, which says that I should get an error message. I did not. When I ran npm start in my village, I saw input boxes for name, age, and height, an "Add to the village" button, and "Smurf Village" written beneath. There was no error, it was just empty.
+
+** Step 2: tell axios to get the smurfs that are already on the server
+*** I did this in my App.js document. The instructions already commented in there were kind of confusing to me. I'm not sure what they meant by passing the right properties on state down to props so I ignored this stuff and eventually deleted it because it was confusing me.
+*** I added an axios request in a componentDidMount thing because in class Dan said that you should always put it in componentDidMount. I am not 100% clear on why but anything in componentDidMount runs as soon as the component is mounted and I guess if it somehow ran before then there would be nothing to set state to, and if it ran afterwards the user would get impatient or something. I don't know. I just know it's a good practice to put it in here.
```

Comment:
The main thing about this is when put something in componentDidMount, this function is ran _just_ before anything is rendered on the screen. 

If you do it after the component renders the whatever you are displaying might be blank and then it updates after.

The user might not get impatient because its a matter of less than a second in a lot of cases. The screen might just jump a bit, which affects user experience (UX)